applications/StructuralMechanicsApplication/custom_strategies/custom_strategies/eigensolver_strategy.hpp

```diff
@@ -650,6 +660,45 @@ class EigensolverStrategy
         KRATOS_CATCH("")
     }
 
+    /// Normalize the eigenmodes with respect to the mass matrix
+    void MassNormalizeEigenvectors(DenseMatrixType& rEigenvectors)
+    {
+        const SparseMatrixType& rMassMatrix = this->GetMassMatrix();
+        const std::size_t num_modes = rEigenvectors.size1();
+        const std::size_t num_dofs  = rEigenvectors.size2();
+
+        DenseVectorType phi(num_dofs);
+        DenseVectorType tmp(num_dofs);
+
+        for (std::size_t j = 0; j < num_modes; ++j) {
+             // copy row j -> phi
+            IndexPartition<std::size_t>(num_dofs).for_each([&](std::size_t i){
```

Comment:
```suggestion
            IndexPartition<std::size_t>(num_dofs).for_each([&phi, &rEigenvectors](std::size_t i){
```

I would capture only what is required.

Reply:
Changed